### PR TITLE
[CALCITE-2322] Add fetch row count support

### PR DESCRIPTION
**Pull request by zacMode**
This resurrects https://github.com/apache/calcite-avatica/pull/49, updating @kminder 's existing work per guidance from [CALCITE-2322](https://issues.apache.org/jira/browse/CALCITE-2322?focusedCommentId=17370941&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-17370941). 

The purpose of this change is to add support for configurable "fetch size"/ "fetch row count". Currently, setting [fetchSize](https://docs.oracle.com/javase/8/docs/api/java/sql/Statement.html#setFetchSize-int-) has no effect, because avatica always uses a hardcoded default value. This PR resolves that issue.

### CODE REVIEW

File: core/src/main/java/org/apache/calcite/avatica/BuiltInConnectionProperty.java
Comment (vlsi) [inline core/src/main/java/org/apache/calcite/avatica/BuiltInConnectionProperty.java:92]: What do you think of `default_fetch_size`?

Comment (zacMode) [reply core/src/main/java/org/apache/calcite/avatica/BuiltInConnectionProperty.java:92]: I would second @julianhyde 's  [feedback here](https://issues.apache.org/jira/browse/CALCITE-2322?focusedCommentId=17370941&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-17370941):

> not "default_fetch_row_count" (because it is understood that this is a default that can be overridden per-statement; furthermore, there is a default default, which is 100);

File: core/src/main/java/org/apache/calcite/avatica/AvaticaStatement.java
Comment (vlsi) [inline core/src/main/java/org/apache/calcite/avatica/AvaticaStatement.java:39]: I would refrain from making unrelated changes

**Review by vlsi — APPROVED** [review]
LGTM modulo the blank line in AvaticaStatement

**Comment by zacMode** [comment]
@F21 @zabetak @vlsi  - any update on when this can make it into a release? Thanks!

**Review by zabetak — COMMENTED** [review]
Overall LGTM, is it possible to add some unit tests verifying the default and overriden values?

**Comment by gabor-hargitai-privitar** [comment]
Hello, what is holding up this feature being merged? It would be very useful.

**Comment by johndww** [comment]
Hi @zabetak / @zacMode . Any idea when this could be merged or if I could help move this through? I ran into this issue (fetch size stuck at 100) as well which isn't working for very large size results over HTTP.

File: core/src/main/java/org/apache/calcite/avatica/BuiltInConnectionProperty.java
Comment (julianhyde) [reply core/src/main/java/org/apache/calcite/avatica/BuiltInConnectionProperty.java:92]: Agree. The property should be called `fetch_size`. JDK has a method 'ResultSet.setFetchSize(int rows)' and so 'fetch size' is unambiguously a row count.

**Pull request closed without merging**
